tests/python_package_test/test_engine.py

```diff
@@ -746,6 +746,14 @@ def test_ranking_prediction_early_stopping():
     with pytest.raises(AssertionError):
         np.testing.assert_allclose(ret_early, ret_early_more_strict)
 
+def test_ranking_with_position_information():
+    rank_example_dir = Path(__file__).absolute().parents[2] / 'examples' / 'lambdarank'
+    params = {
+        'objective': 'lambdarank',
+        'verbose': -1
+    }
+    lgb_train = lgb.Dataset(str(rank_example_dir / 'rank.train'), params=params)
+    gbm = lgb.train(params, lgb_train, num_boost_round=50)
```

Comment:
Please add some assertions checking that LightGBM is actually working correctly. For example, checking that it provides good performance on the training data or a holdout set.

This test is just checking that it runs without raising any errors, which is not a very strong protection against future issues or strong evidence that these changes don't break existing functionality.

Reply:
The change doesn't affect the existing functionality when the positional information is not provided (which should be covered by existing test cases). And the aim of this test case is to ensure the positional information is loaded properly when provided and doesn't break anything. We will later provide a rigorous performance comparison of the proposed method using corresponding datasets with position bias in the relevance labels.

Reply:
> We will later provide a rigorous performance comparison of the proposed method using corresponding datasets with position bias in the relevance labels.

"later" as in "after this PR is merged"?

I personally am not comfortable with LightGBM taking on this new code and then the work being done LATER to confirm that it improves LightGBM's performance and add tests that it works correctly.

Changes in behavior should be accompanied by meaningful tests verifying that those changes work as expected.

Reply:
What about comparing an evaluation metric of the trained model to a fixed value? For example, the NDCG on train or valid set should be above some threshold.

Reply:
That would be great! Just something to confirm that it's working as expected, to ensure the newly-added code is actually run when the tests are run, and to help us catch when changes break this code.

Reply:
I added a test case to ensure that the performance of the baseline (no positions specified) does not differ much from the case when we specify the randomly assigned positions. Would this suffice?